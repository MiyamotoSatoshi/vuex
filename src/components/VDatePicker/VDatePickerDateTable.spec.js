import { compileToFunctions } from 'vue-template-compiler'
import VDatePickerDateTable from './VDatePickerDateTable'
import { test } from '@util/testing'

test('VDatePickerDateTable.js', ({ mount }) => {
  it('should render component and match snapshot', () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05',
        current: '2005-07',
        value: '2005-11-03'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with events and match snapshot', () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05',
        current: '2005-07',
        value: '2005-11-03',
        events: ['2005-05-03'],
        eventColor: 'red'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with events colored by object and match snapshot', () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05',
        current: '2005-07',
        value: '2005-11-03',
        events: ['2005-05-03', '2005-05-04'],
        eventColor: {'2005-05-03': 'red', '2005-05-04': 'blue lighten-1'}
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render component with events colored by object and match snapshot', () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05',
        current: '2005-07',
        value: '2005-11-03',
        events: ['2005-05-03', '2005-05-04'],
        eventColor: date => ({'2005-05-03': 'red'}[date])
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with first day of week', function () {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05',
        current: '2005-07',
        value: '2005-11-03',
        firstDayOfWeek: 2
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should watch tableDate value and run transition', async () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05',
        current: '2005-07',
        value: '2005-11-03'
      }
    })

    wrapper.setProps({
      tableDate: '2005-06'
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.find('table')[0].element.className).toBe('tab-transition-enter tab-transition-enter-active')
  })

  it('should watch tableDate value and run reverse transition', async () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05',
        current: '2005-07',
        value: '2005-11-03'
      }
    })

    wrapper.setProps({
      tableDate: '2005-04'
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.find('table')[0].element.className).toBe('tab-reverse-transition-enter tab-reverse-transition-enter-active')
  })

  it('should emit event when date button is clicked', () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05',
        current: '2005-07',
        value: '2005-11-03'
      }
    })

    const input = jest.fn()
    wrapper.vm.$on('input', input)

    wrapper.find('tbody button')[0].trigger('click')
    expect(input).toBeCalledWith('2005-05-01')
  })

  it('should not emit event when disabled month button is clicked', () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05',
        current: '2005-07',
        value: '2005-11-03',
        allowedDates: () => false
      }
    })

    const input = jest.fn()
    wrapper.vm.$on('input', input)

    wrapper.find('tbody button')[0].trigger('click')
    expect(input).not.toBeCalled()
  })

  it('should emit tableDate event when scrolled and scrollable', () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05',
        scrollable: true
      }
    })

    const tableDate = jest.fn()
    wrapper.vm.$on('tableDate', tableDate)

    wrapper.trigger('wheel')
    expect(tableDate).toBeCalledWith('2005-06')
  })

  it('should not emit tableDate event when scrolled and not scrollable', () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05'
      }
    })

    const tableDate = jest.fn()
    wrapper.vm.$on('tableDate', tableDate)

    wrapper.trigger('wheel')
    expect(tableDate).not.toBeCalled()
  })

  // TODO
  it.skip('should emit tableDate event when swiped', () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05'
      }
    })

    const tableDate = jest.fn()
    wrapper.vm.$on('tableDate', tableDate)

    wrapper.trigger('touchstart')
    wrapper.trigger('touchend')
    expect(tableDate).toBeCalledWith('2005-06')
  })

  it('should return null for selected date if not provided', () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05'
      }
    })

    expect(wrapper.vm.selectedDate).toBe(null)
    expect(wrapper.vm.selectedMonth).toBe(null)
    expect(wrapper.vm.selectedYear).toBe(null)
  })

  it('should calculate active date', () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05',
        value: '2005-05-11'
      }
    })

    expect(wrapper.vm.isSelected(11)).toBe(true)
    expect(wrapper.vm.isSelected(12)).toBe(false)
  })

  it('should calculate active date', () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05',
        current: '2005-05-11'
      }
    })

    expect(wrapper.vm.isCurrent(11)).toBe(true)
    expect(wrapper.vm.isCurrent(12)).toBe(false)
  })

  it('should change tableDate when touch is called', () => {
    const wrapper = mount(VDatePickerDateTable, {
      propsData: {
        tableDate: '2005-05'
      }
    })

    const tableDate = jest.fn()
    wrapper.vm.$on('tableDate', tableDate)

    wrapper.vm.touch(1)
    expect(tableDate).toBeCalledWith('2005-06')
    wrapper.vm.touch(-1)
    expect(tableDate).toBeCalledWith('2005-04')
  })
})
