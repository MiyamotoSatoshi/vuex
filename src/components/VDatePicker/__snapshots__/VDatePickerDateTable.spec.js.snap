// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VDatePickerDateTable.js should match snapshot with first day of week 1`] = `

<div class="date-picker-table date-picker-table--date">
  <table>
    <thead>
      <tr>
        <th>
          T
        </th>
        <th>
          W
        </th>
        <th>
          T
        </th>
        <th>
          F
        </th>
        <th>
          S
        </th>
        <th>
          S
        </th>
        <th>
          M
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
        </td>
        <td>
        </td>
        <td>
        </td>
        <td>
        </td>
        <td>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              1
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              2
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              3
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              4
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              5
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              6
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              7
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              8
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              9
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              10
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              11
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              12
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              13
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              14
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              15
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              16
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              17
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              18
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              19
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              20
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              21
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              22
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              23
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              24
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              25
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              26
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              27
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              28
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              29
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              30
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              31
            </div>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>

`;

exports[`VDatePickerDateTable.js should render component and match snapshot 1`] = `

<div class="date-picker-table date-picker-table--date">
  <table>
    <thead>
      <tr>
        <th>
          S
        </th>
        <th>
          M
        </th>
        <th>
          T
        </th>
        <th>
          W
        </th>
        <th>
          T
        </th>
        <th>
          F
        </th>
        <th>
          S
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              1
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              2
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              3
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              4
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              5
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              6
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              7
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              8
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              9
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              10
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              11
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              12
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              13
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              14
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              15
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              16
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              17
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              18
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              19
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              20
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              21
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              22
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              23
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              24
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              25
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              26
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              27
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              28
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              29
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              30
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              31
            </div>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>

`;

exports[`VDatePickerDateTable.js should render component with events and match snapshot 1`] = `

<div class="date-picker-table date-picker-table--date">
  <table>
    <thead>
      <tr>
        <th>
          S
        </th>
        <th>
          M
        </th>
        <th>
          T
        </th>
        <th>
          W
        </th>
        <th>
          T
        </th>
        <th>
          F
        </th>
        <th>
          S
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              1
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              2
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              3
            </div>
          </button>
          <div class="date-picker-table__event red">
          </div>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              4
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              5
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              6
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              7
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              8
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              9
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              10
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              11
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              12
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              13
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              14
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              15
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              16
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              17
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              18
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              19
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              20
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              21
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              22
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              23
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              24
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              25
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              26
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              27
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              28
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              29
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              30
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              31
            </div>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>

`;

exports[`VDatePickerDateTable.js should render component with events colored by object and match snapshot 1`] = `

<div class="date-picker-table date-picker-table--date">
  <table>
    <thead>
      <tr>
        <th>
          S
        </th>
        <th>
          M
        </th>
        <th>
          T
        </th>
        <th>
          W
        </th>
        <th>
          T
        </th>
        <th>
          F
        </th>
        <th>
          S
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              1
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              2
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              3
            </div>
          </button>
          <div class="date-picker-table__event red">
          </div>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              4
            </div>
          </button>
          <div class="date-picker-table__event blue lighten-1">
          </div>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              5
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              6
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              7
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              8
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              9
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              10
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              11
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              12
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              13
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              14
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              15
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              16
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              17
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              18
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              19
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              20
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              21
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              22
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              23
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              24
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              25
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              26
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              27
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              28
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              29
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              30
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              31
            </div>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>

`;

exports[`VDatePickerDateTable.js should render component with events colored by object and match snapshot 2`] = `

<div class="date-picker-table date-picker-table--date">
  <table>
    <thead>
      <tr>
        <th>
          S
        </th>
        <th>
          M
        </th>
        <th>
          T
        </th>
        <th>
          W
        </th>
        <th>
          T
        </th>
        <th>
          F
        </th>
        <th>
          S
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              1
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              2
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              3
            </div>
          </button>
          <div class="date-picker-table__event red">
          </div>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              4
            </div>
          </button>
          <div class="date-picker-table__event accent">
          </div>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              5
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              6
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              7
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              8
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              9
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              10
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              11
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              12
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              13
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              14
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              15
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              16
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              17
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              18
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              19
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              20
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              21
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              22
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              23
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              24
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              25
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              26
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              27
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              28
            </div>
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              29
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              30
            </div>
          </button>
        </td>
        <td>
          <button type="button"
                  class="btn btn--flat btn--floating"
          >
            <div class="btn__content">
              31
            </div>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>

`;
